gossipd/gossipd.c

```diff
@@ -350,8 +350,10 @@ static void handle_remote_addr(struct daemon *daemon, const u8 *msg)
 	if (!fromwire_gossipd_remote_addr(msg, &remote_addr))
 		master_badmsg(WIRE_GOSSIPD_REMOTE_ADDR, msg);
 
-	/* current best guess is that we use DEFAULT_PORT on public internet */
-	remote_addr.port = DEFAULT_PORT;
+	/* We can only guess the public port here, so we use DEFAULT_PORT.
+	 * Also the DEFAULT_PORT differs on which network is used.
+	 * So we do it similar to what Bitcoin Core does */
+	remote_addr.port = DEFAULT_PORT + chainparams->rpc_port - 8332;
```

Comment:
We should keep our eyes on https://github.com/ElementsProject/lightning/pull/5242 will be a new chain value of the default port, and remove these tricky calculation :)